src/__tests__/render.test.tsx

```diff
@@ -285,3 +281,160 @@ test('RenderAPI type', () => {
   render(<Banana />) as RenderAPI;
   expect(true).toBeTruthy();
 });
+
+test('should throw when rendering a string outside a text component', () => {
+  expect(() =>
+    render(<View>hello</View>, {
+      unstable_validateStringsRenderedInText: true,
+    })
+  ).toThrowErrorMatchingInlineSnapshot(
+    `"Text strings must be rendered within a host Text component. Text hello was rendered in a View"`
+  );
+});
+
+const profilerErrorMessage =
+  'The above error occurred in the <Profiler> component';
+
+test('should throw an error when rerendering with text outside of Text component', () => {
+  // eslint-disable-next-line no-console
+  console.error = (errorMessage: string) => {
+    if (!errorMessage.includes(profilerErrorMessage)) {
+      originalConsoleError(errorMessage);
+    }
+  };
+  const { rerender } = render(<View />, {
+    unstable_validateStringsRenderedInText: true,
+  });
+
+  expect(() => rerender(<View>hello</View>)).toThrowError(
+    'Text strings must be rendered within a host Text component.'
+  );
+  // eslint-disable-next-line no-console
+  console.error = originalConsoleError;
+});
+
+const ErrorComponent = () => {
+  const [shouldDisplayText, setShouldDisplayText] = React.useState(false);
+
+  if (!shouldDisplayText) {
+    return (
+      <Pressable
+        onPress={() => {
+          setShouldDisplayText(true);
+        }}
+      >
+        <Text>Display text</Text>
+      </Pressable>
+    );
+  }
+
+  return <View>text rendered outside text component</View>;
+};
+
+test('should throw an error when strings are rendered outside Text', () => {
+  // eslint-disable-next-line no-console
+  console.error = (errorMessage: string) => {
+    if (!errorMessage.includes(profilerErrorMessage)) {
+      originalConsoleError(errorMessage);
+    }
+  };
+  const { getByText } = render(<ErrorComponent />, {
+    unstable_validateStringsRenderedInText: true,
+  });
+
+  expect(() => fireEvent.press(getByText('Display text'))).toThrowError(
+    'Text strings must be rendered within a host Text component.'
```

Comment:
Why does this message doesn't contain string + node type like the other tests (`"Text strings must be rendered within a host Text component. Text 0 was rendered in a View"`)?

Reply:
Oh I forgot to replace it with an inline snapshot, toThrowErrror actually works when the tested string is included in the error message so this works even though the actual message is more complete